os/arch/arm/src/common/up_checkspace.c

```diff
@@ -32,17 +32,43 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
-extern uint32_t _stext;
-extern uint32_t _etext;
+extern uint32_t _stext_flash;
+extern uint32_t _etext_flash;
 extern uint32_t _sdata;
 extern uint32_t _edata;
 extern uint32_t _sbss;
 extern uint32_t _ebss;
+#ifdef CONFIG_ARCH_HAVE_RAM_KERNEL_TEXT
+extern uint32_t _stext_ram;
+extern uint32_t _etext_ram;
+#endif
+
+bool is_kernel_text_space(void *addr)
+{
+	/* Check if the address lies in the kernel text section range */
+	if ((addr >= (void *)&_stext_flash && addr <= (void *)&_etext_flash)
+#ifdef CONFIG_ARCH_HAVE_RAM_KERNEL_TEXT
+		|| (addr >= (void *)&_stext_ram && addr <= (void *)&_etext_ram)
+#endif
+	) {
+		return true;
+	}
+	return false;
+}
+
+bool is_kernel_data_space(void *addr)
+{
+	/* Check if the address lies in the kernel data section range */
+	if (addr >= (void *)&_sdata && addr <= (void *)&_ebss) {
```

Comment:
This is valid when bss is located at next of data section. But can we guarantee that or should we layout like that always? I don't think so. So, how about separating bss and data?
```c
if ((addr >= (void *)&_sdata && addr <= (void *)&_edata) || (addr >= (void *)&_sbss && addr <= (void *)&_ebss)) {
```